gradle/module-validation.gradle

```diff
@@ -0,0 +1,55 @@
+import groovy.json.JsonSlurper
+
+/*
+ * This buildscript contains tasks related to the validation of each module in fabric api.
+ *
+ * Right now this task verifies each Fabric API module has a module lifecycle specified.
+ * More functionality will probably be added in the future.
+ */
+
+subprojects {
+	// Create the task
+	task validateModules(type: ValidateModuleTask)
+}
+
+/**
+ * Verifies that each module has the required custom values for module lifecycle in it's FMJ.
+ *
+ * <p>Example:
+ * <pre>{@code
+ * "custom": {
+ *   "fabric-api:module-lifecycle": "stable"
+ * }
+ * }</pre>
+ */
+class ValidateModuleTask extends DefaultTask {
+	ValidateModuleTask() {
+		group = "verification"
+
+		// Hook up validation to check task
+		project.tasks.check.dependsOn(this)
+	}
+
+	@TaskAction
+	void validate() {
+		def json = new JsonSlurper().parse(project.file("src/main/resources/fabric.mod.json") as File) as Map<String, Map<String, String>>;
+
+		if (json.custom == null) {
+			throw new GradleException("Module ${project} does not have a custom value containing module lifecycle!")
+		}
+
+		if (json.custom.get("fabric-api:module-lifecycle") == null) {
```

Comment:
I wonder how costly calling this is, but I personally prefer storing to local var to avoid extra map lookups

Reply:
I believe JsonSlurper serializes the file into the respective `Map`, `List`, `Number`, `Boolean`, `null`s for the things in json